### ISSUE TITLE
feat(myLocation): call MyLocationButton programmatically

### ISSUE THREAD

**Issue by DwieDima** [capacitor-community/capacitor-googlemaps-native#136]
**!! This feature request is related to the fork of @tafelnl !!**

**Is your feature request related to a problem? Please describe.**
Currently it is possible to display the native myLocationButton in the map, which allows navigation to the current location.
At the moment, programmatic navigation is not possible, which excludes two important use-cases:

1. Navigation to the current location, due to life-cycle events or a specific business logic.
2. Creating custom UI to trigger navigation to the current location (e.g. using `ion-fab` instead of MyLocationButton).

A workaround would be the method `moveCamera()` by querying the current location beforehand.
However, this loses the advantage of the event listener `didTapMyLocationButton()`.

**Describe the solution you'd like**
Extend CapacitorGoogleMaps with the method `tapMyLocationButton()`, which triggers the click on myLocationButton.

**Comment by tafelnl** [comment]
Thanks for opening this issue. I am not aware of such a feature in either the iOS or the Android SDK. We will divert from the 
"standard" API significantly by implementing this . I think Google might have good reasons not to implement this into their SDKs. So I don't want to divert too much. But maybe I overlooked something in the docs and it actually _does_ exist. If that's the case, could you please share the link?

**Comment by tafelnl** [comment]
Thanks for opening this issue. I am not aware of such a feature in either the iOS or the Android SDK. We will divert from the 
"standard" API significantly by implementing this . I think Google might have good reasons not to implement this into their SDKs. So I don't want to divert too much. But maybe I overlooked something in the docs and it actually _does_ exist. If that's the case, could you please share the link?

Of course this use case is special since in the Android or iOS SDK, it would typically not happen that you'd have an overlay HTMLElement in place. But still it would be possible in Android and iOS to remove the default myLocationButton and replace it with your own as well. And even then you would still need to do the custom logic you described above as well.

What do you think?

**Comment by DwieDima** [comment]
I quickly did some research and also did not find such a method natively.
However, I found some similar feature requests for various frameworks like flutter.
According to this [comment](https://github.com/flutter/flutter/issues/29753#issuecomment-513091166), MyLocationButton is natively created by the SDK and can't be edited either.
The correct approach here would actually also be a manual animation to the current location.

I found the implementation on stackoverflow, where for [IOS](https://stackoverflow.com/a/41041393) and [Android](https://stackoverflow.com/a/23926705/18223464) only a few lines of code are needed.
At this point, of course, the question arises whether this "workaround" is performed in the client, or implemented in the plugin.

In your [documentation](https://dutchconcepts.github.io/capacitor-google-maps/#/api) I have not yet found a method that allows retrieving the current location (correct me here if I'm wrong).

For me as a enduser of this plugin, implementing this to the plugin brings of course a lot of comfort.
If you choose not to add this to the plugin, a method for retrieving the current location must be provided.

Whats your opinion about that?

**Comment by tafelnl** [comment]
> In your [documentation](https://dutchconcepts.github.io/capacitor-google-maps/#/api) I have not yet found a method that allows retrieving the current location (correct me here if I'm wrong).

Yeah that's the thing. Retrieving the location should be done by using the [offical Capacitor geolocation plugin](https://www.npmjs.com/package/@capacitor/geolocation). It would not make sense to include it in this plugin. But since adding a feature like `tapMyLocationButton` actually requires adding such functionality, it is a bit contradictory.

Maybe I could add this functionality on the JavaScript side only, and then make `@capacitor/geolocation` a dependency for that specific method. So with a bundler, like webpack, you would still profit from correct treeshaking and not unnecessary adding `@capacitor/geolocation` when it's not needed.

However, this would require the Capacitor team (or anyone for that matter) to first answer my open question here: https://github.com/ionic-team/capacitor/discussions/5429

**Comment by va2ron1** [comment]
@tafelnl doing this [experiment](https://github.com/capacitor-community/capacitor-googlemaps-native/discussions/137), all method execution from the web is handle by Capacitor to the plugin using the Two-Way communication system of the browser (postMessage). There's not actual parsing, everything is converted to a Dictionary. ([Reference](https://github.com/ionic-team/capacitor/blob/55bf004897b3feb280ab6b6575d2a2c1a0a183e2/ios/Capacitor/Capacitor/WebViewDelegationHandler.swift#L173)). In this [point ](https://github.com/ionic-team/capacitor/blob/55bf004897b3feb280ab6b6575d2a2c1a0a183e2/ios/Capacitor/Capacitor/WebViewDelegationHandler.swift#L155) you can parse and get the element, but there's no easy way to overwrite this method.
